R/unneeded_concatenation_linter.R

```diff
@@ -9,6 +9,39 @@
 #'   [as.vector()] is not always preferable, for example with environments
 #'   (especially, `R6` objects), in which case `list()` is the better alternative.
 #'
+#' @examples
+#' # will produce lints
+#' lint(
+#'   text = "x <- c()",
+#'   linters = unneeded_concatenation_linter()
+#' )
+#'
+#' lint(
+#'   text = "x <- c(TRUE)",
+#'   linters = unneeded_concatenation_linter()
+#' )
+#'
+#' lint(
+#'   text = "c(a:b)",
+#'   linters = unneeded_concatenation_linter(allow_single_expression = FALSE)
+#' )
+#'
+#' # okay
+#' lint(
+#'   text = "x <- NULL",
```

Comment:
I would also present the alternative of x <- integer(10) for cases where c() was meant to seed a vector of known size

Reply:
I think it warrants a comment on the example to say why it's there, otherwise there's a bit of a "what's this example got to do w anything?" reaction

Reply:
Good point. Added a comment.